src/h5web/providers/hsds/api.ts

```diff
@@ -163,31 +164,33 @@ export class HsdsApi implements ProviderAPI {
 
     this.datasets[id] = {
       collection: HDF5Collection.Datasets,
+      name,
       ...dataset,
       ...(attributes ? { attributes } : {}),
     };
   }
 
-  private async processDatatype(id: HDF5Id): Promise<void> {
+  private async processDatatype(id: HDF5Id, name: string): Promise<void> {
     const datatype = await this.fetchDatatype(id);
 
     this.datatypes[id] = {
       collection: HDF5Collection.Datatypes,
+      name,
       ...datatype,
     };
   }
 
   private async resolveLink(link: HDF5HardLink | HDF5RootLink): Promise<void> {
-    const { collection, id } = link;
+    const { collection, id, title } = link;
 
     if (collection === HDF5Collection.Groups) {
-      return this.processGroup(id);
+      return this.processGroup(id, title);
     }
     if (collection === HDF5Collection.Datasets) {
-      return this.processDataset(id);
+      return this.processDataset(id, title);
     }
     if (collection === HDF5Collection.Datatypes) {
-      return this.processDatatype(id);
+      return this.processDatatype(id, title);
```

Comment:
In HSDS, the `name` of the entity is gotten from the title of the link pointing to the entity.